lib/problem_check/twitter_login.rb

```diff
@@ -2,19 +2,22 @@
 
 class ProblemCheck::TwitterLogin < ProblemCheck
   self.priority = "high"
-
-  # TODO: Implement.
   self.perform_every = 24.hours
+  self.max_blips = 3
 
-  def call
+  def call(tracker)
     return no_problem if !authenticator.enabled?
     return no_problem if authenticator.healthy?
 
+    authenticator.disable if disable_failing_social_logins? && self.class.max_blips >= tracker.blips
+
     problem
   end
 
   private
 
+  delegate :disable_failing_social_logins?, to: SiteSetting, private: true
```

Comment:
Curious why you added this? I am not sure it is necessary personally instead of just calling `SiteSetting.disable_failing_social_logins?` inline

Reply:
Probably a bit premature. The reason is if we wanted to change the implementation of `#disable_failing_social_logins?` to a local method this would be an entirely trivial affair. But I agree with you this is gratuitous.